### ISSUE TITLE
nerfing the Melee scale

### ISSUE THREAD

**Issue by Hng-Vit** [Globox1997/LevelZ#371]
### Make sure you are not opening a duplicate.

- [X] I have searched the issue tracker and did not find an issue describing my bug/feature.

### Description of your suggestion.

after maxing the Melee skill, i can kill nearly everything with barehand and it's quite boring. So how can i adjust the stats of the skills or can u add another section in the mod settings? I know adding another feature will cost a lot of time, so just take this as a small vision.

**Comment by 707peter** [comment]
yeah this mod is far from finished and i don't know why so simple things like this haven't been done.
like we can reduce the bow extra damage in game or in the files but not the base damage increase. 
its soo stupid to run around with 300 damage mele and 40 blocks of range....

**Comment by Hng-Vit** [comment]
the most annoying thing is the block range, man it ruins everything

**Comment by Globox1997** [comment]
Working on it.

**Comment by h4x0reL** [comment]
Killing a Warden in 3 hits is indeed boring. As a temporary solution I suggest downloading the RpgDifficulty mod so that the mobs will be stronger over time.

**Comment by Globox1997** [comment]
Probably fixed with the latest version.

**Comment by sirrobzeroone** [comment]
Hi @Globox1997 is there a new setting that we can add to the levelz config file so as to adjust the melee damage multiplier per level?

**Comment by Globox1997** [comment]
Just update to the latest version.
You can adjust it by a datapack

**Comment by Hng-Vit** [comment]
which data pack it is?

**Comment by sirrobzeroone** [comment]
I'm not familiar with MC modding or Fabric, I just did a few minutes of reading and code digging and I believe we just need to create a datapack file as per this ref: https://minecraft.wiki/w/Data_pack & https://github.com/Globox1997/LevelZ/tree/bfc0512c353c7f03446a2aeb75f60cedf407f20b?tab=readme-ov-file#datapacks
and adjust the values that are stored in default.json via that datapack (https://github.com/Globox1997/LevelZ/blob/bfc0512c353c7f03446a2aeb75f60cedf407f20b/src/main/resources/data/levelz/skill/default.json#L23) -

`` "melee": {
    "replace": false,
    "id": 1,
    "key": "melee",
    "level": 20,
    "attributes": [
      {
        "id": 1,
        "type": "generic.attack_damage",
        "operation": "ADD_VALUE",
        "value": 0.2
      },
      {
        "type": "generic.attack_knockback",
        "operation": "ADD_VALUE",
        "value": 0.05
      },
      {
        "type": "player.sweeping_damage_ratio",
        "base": 0,
        "operation": "ADD_VALUE",
        "value": 0.03
      }
    ],``

I think it would be the "generic.attack_damage - value" initially to adjust, but I find knockback also gets slightly OP as well although super fun when they knockback on death and fly off a dozen blocks or so.

Hopefully if I'm way off the mark someone with more knowledge in this space can correct my understanding above.

**Comment by sirrobzeroone** [comment]
I gave it a shot, but no luck, my pack.mcmeta loads just fine but either my subfoldering below that or something inside the JSON is bad, not sure which.

Just in case this is helpful to someone with more expertise my folder structure and file contents are below - I'm probably missing something basic but very fundamental, appreciate any help.

-- WorldName
--- datapacks
---- levelz_custom
----- pack.mcmeta
----- data
------ levelz (tried levelz_custom)
------- skill
-------- melee.json

pack.mcmeta content
```
{
    "pack": {
        "pack_format": 48,
        "description": "LevelZ Custom Melee Damage"
    }
}
```

melee.json content
```
{
  "melee": {
    "replace": false,
    "id": 1,
    "key": "melee",
    "level": 20,
    "attributes": [
      {
        "id": 1,
        "type": "generic.attack_damage",
        "operation": "ADD_VALUE",
        "value": 0.1
      },
      {
        "type": "generic.attack_knockback",
        "operation": "ADD_VALUE",
        "value": 0.025
      },
      {
        "type": "player.sweeping_damage_ratio",
        "base": 0,
        "operation": "ADD_VALUE",
        "value": 0.015	
      }
    ],
    "bonus": [
      {
        "level": 1,
        "key": "meleeCriticalAttackDamage"
      },
      {
        "level": 15,
        "key": "meleeDoubleAttackDamageChance"
      }
    ]
  }
}
```

**Comment by Globox1997** [comment]
Set replace true, than it should work

**Comment by sirrobzeroone** [comment]
Thanks for the help much appreciated - gave it a try and no luck. I'm using 1.21.1, with fabric 0.16.9, fresh world with just fabric, cloth, libz and levelz as mods active. I've been using these two commands to add and remove to check if the multiplyer adjusts, just did 5 for no real reason plus some /reloads and /datapack list enabled to ensure everythings running :)

```
/level @s add melee 5
/level @s remove melee 5
```

Thanks again for your time helping with this.

EDIT-1: Just to make sure the pack was fully loading I added a new 2 stick crafting recipe for a wooden sword and after a bit of fiddling around it loaded in fine - Just wanted to confirm 100% the pack was loading and this seemed the easy way to confirm.

EDIT-2:  I gave 1.20.1, with fabric 0.15.6 a go and the correct versions for that of Levelz, cloth, fabric api and libz. Minor tweak and my 2 stick wooden sword recipe works okay but still getting melee increments of 0.2 rather than 0.1.

**Comment by sirrobzeroone** [comment]
I was skimming the issues and noticed this new one by @RomanKnight - #379, seemed similar so copied the entire default.json contents into a new file called default_new.json and then disabled the default.json skills using modmenu and updated just those 3 melee values in my new file. This then allowed my custom 0.1 to load in and be used.

Not sure thats quite what was intended in regards to usage but its a workaround that works. I'll do some further tinkering/playing around. 

Thank you for your time keep these mods updated I know how time consuming that is.

**Comment by sirrobzeroone** [comment]
Hmm adjusting that value has minimal impact on the damage amount, the huge addition looks to be driven from the character something - ?level? - hard for me to work out.

EDIT: I might have to leave this here for the moment, I have done some testing and at low levels the damage bonus seems to work as I would expect ie skill lvl * generic.attack_damage:value. However at some Character Level (not skill) this changes and Character something ?Level? seems to end up in the mix. I havent determined that point from testing and I cant read javascript code well enough to understand whats going on in the code. My tests:

My Level 4 Character with melee skill at 12 gets a very simple - sword damage(4 for wooden) + Skill damage (2.4 unchanged) = 6.4 
![image](https://github.com/user-attachments/assets/c18ec354-63c1-4a87-a84f-f91767c89260)

My Level 56 Character with melee skill at 15 seems to get - sword damage + skill damage + about 300 from somewhere which also goes an extra decmial place down = 316.82
![image](https://github.com/user-attachments/assets/c5e1fa8a-904f-482d-8b31-bc8149ce7228)

Whatever the 300 is coming from seems to provide an unbalanced shift compared to the other values ie weapon and melee skill bonus.

**Comment by sirrobzeroone** [comment]
Sorry just want to save @Globox1997 sometime if possible - I was trying to replicate the issue using levelz experience globes and I couldn't, I did this lvl by lvl aand spent skill points to try and keep it realer but as can be seen no mystery +300.

Same mod mix, only difference was my image above is from my server and was done the old fashioned way slaying beasties were as the below is a local game and used experience orbs.

![image](https://github.com/user-attachments/assets/4a8828b6-caff-4b4d-b573-f25d305c3516)

I checked the two player save files but they are very similar, XpTotal was much more on the server version but given I only used levelz experience orbs that fits, Score was a heap lower not sure if thats a Levelz tracked metric? Not sure if theres some sort of saved player metrics in the world file, happy to check if it would help.

EDIT: left server dat file right local dat.....that explains some of the random addition, I'll try and test more may not even be LevelZ :
![image](https://github.com/user-attachments/assets/33abfa90-63f5-429d-8f1f-b01ca766a31d)

**Comment by sirrobzeroone** [comment]
Since I fixed the player dat files we have been playing pretty solid ona server with 4 players off and on and so far no ones attack damage has increased like the above. i've tried replicating a few times ona solo game with just levelz and likewise I've been unable to replicate the issue. Not sure if the slightly newer version of levelz fixed the issue or it was somethng else that caused the problem but wanted to provide an update.

**Comment by Globox1997** [comment]
This should be definitely resolved with the latest version :)
